internal/cephfs/controllerserver.go

```diff
@@ -689,6 +689,22 @@ func (cs *ControllerServer) DeleteSnapshot(ctx context.Context, req *csi.DeleteS
 			}
 			return &csi.DeleteSnapshotResponse{}, nil
 		}
+		// if the error is ErrVolumeNotFound, delete the snapshot as the
+		// snapshot and subvolume is already deleted
+		if errors.Is(err, ErrVolumeNotFound) {
+			util.ErrorLog(ctx, "Volume not present")
+			err = volOpt.deleteSnapshot(ctx, volumeID(sid.FsSnapshotName), volumeID(sid.FsSubvolName))
```

Comment:
question: is it possible to delete a snapshot of the subvolume that is already deleted?

Reply:
No, here we are already deleting the subvolume and snapshot from backend. Hence,  we just need to check for the omap entries.